### ISSUE TITLE
Failed build when using AGP 8.1

### ISSUE THREAD

**Issue by evgfilim1** [4Q-s-r-o/ota_update#108]
## Description

Cannot use the plugin with AGP (Android Gradle Plugin) 8.1 due to "Namespace not specified" error from the plugin.

## System info

<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel beta, 3.16.0-0.3.pre, on Arch Linux 6.5.9-arch2-1, locale en_US.UTF-8)
    • Flutter version 3.16.0-0.3.pre on channel beta at /opt/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 476aa717cd (4 weeks ago), 2023-10-19 20:10:52 -0500
    • Engine revision 91cde06da0
    • Dart version 3.2.0 (build 3.2.0-210.3.beta)
    • DevTools version 2.28.1

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /data/distr/android-sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /data/distr/android-sdk
    • Java binary at: /home/evgfilim1/.local/share/JetBrains/Toolbox/apps/android-studio/jbr/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.8.1+0-17.0.8.1b1000.32-11071527)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • CHROME_EXECUTABLE = /usr/bin/chromium

[✓] Linux toolchain - develop for Linux desktop
    • clang version 16.0.6
    • cmake version 3.27.7
    • ninja version 1.11.1
    • pkg-config version 1.8.1

[✓] Android Studio (version 2023.1)
    • Android Studio at /home/evgfilim1/.local/share/JetBrains/Toolbox/apps/android-studio-2
    • Flutter plugin version 76.3.3
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)

[✓] Android Studio (version 2023.2)
    • Android Studio at /home/evgfilim1/.local/share/JetBrains/Toolbox/apps/android-studio
    • Flutter plugin version 76.3.4
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.8.1+0-17.0.8.1b1000.32-11071527)

[✓] IntelliJ IDEA Ultimate Edition (version 2023.2)
    • IntelliJ at /home/evgfilim1/.local/share/JetBrains/Toolbox/apps/intellij-idea-ultimate
    • Flutter plugin version 76.3.4
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] Connected device (3 available)
    • Pixel 7 (mobile) • adb-***********._adb-tls-connect._tcp • android-arm64  • Android 14 (API 34)
    • Linux (desktop)  • linux                                 • linux-x64      • Arch Linux 6.5.9-arch2-1
    • Chrome (web)     • chrome                                • web-javascript • Chromium 119.0.6045.105 Arch Linux

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

</details>

## Steps to reproduce

1. Create a new application flutter project
2. Upgrade gradle wrapper to version 8.4 by changing `distributionUrl` in `android/gradle/wrapper/gradle-wrapper.properties`
3. Upgrade `com.android.application` plugin to version 8.1.3 by changing it in `android/settings.gradle`
4. Execute `flutter run`

## Expected result

Application compiles successfully

## Actual result

```
FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':ota_update'.
> Could not create an instance of type com.android.build.api.variant.impl.LibraryVariantBuilderImpl.
   > Namespace not specified. Specify a namespace in the module's build file. See https://d.android.com/r/tools/upgrade-assistant/set-namespace for information about setting the namespace.

     If you've specified the package attribute in the source AndroidManifest.xml, you can use the AGP Upgrade Assistant to migrate to the namespace value in the build file. Refer to https://d.android.com/r/tools/upgrade-assistant/agp-upgrade-assistant for general information about using the AGP Upgrade Assistant.
```

**Comment by evgfilim1** [comment]
There's an error about R8 unable to run properly on AGP 8.0+, which is run on release builds. The fix is to update okhttp to at least 4.11.0. I'll update my PR.

---

## Steps to reproduce

1. Same as above 1–3
2. Execute `flutter run --release`

## Expected result

Application compiles successfully

## Actual result

```
ERROR: Missing classes detected while running R8. Please add the missing classes or apply additional keep rules that are generated in /home/***/build/app/outputs/mapping/release/missing_rules.txt.
ERROR: R8: Missing class org.bouncycastle.jsse.BCSSLParameters (referenced from: void okhttp3.internal.platform.BouncyCastlePlatform.configureTlsExtensions(javax.net.ssl.SSLSocket, java.lang.String, java.util.List) and 1 other context)
Missing class org.bouncycastle.jsse.BCSSLSocket (referenced from: void okhttp3.internal.platform.BouncyCastlePlatform.configureTlsExtensions(javax.net.ssl.SSLSocket, java.lang.String, java.util.List) and 5 other contexts)
Missing class org.bouncycastle.jsse.provider.BouncyCastleJsseProvider (referenced from: void okhttp3.internal.platform.BouncyCastlePlatform.<init>())
Missing class org.conscrypt.Conscrypt$Version (referenced from: boolean okhttp3.internal.platform.ConscryptPlatform$Companion.atLeastVersion(int, int, int))
Missing class org.conscrypt.Conscrypt (referenced from: boolean okhttp3.internal.platform.ConscryptPlatform$Companion.atLeastVersion(int, int, int) and 4 other contexts)
Missing class org.conscrypt.ConscryptHostnameVerifier (referenced from: okhttp3.internal.platform.ConscryptPlatform$DisabledHostnameVerifier)
Missing class org.openjsse.javax.net.ssl.SSLParameters (referenced from: void okhttp3.internal.platform.OpenJSSEPlatform.configureTlsExtensions(javax.net.ssl.SSLSocket, java.lang.String, java.util.List))
Missing class org.openjsse.javax.net.ssl.SSLSocket (referenced from: void okhttp3.internal.platform.OpenJSSEPlatform.configureTlsExtensions(javax.net.ssl.SSLSocket, java.lang.String, java.util.List) and 1 other context)
Missing class org.openjsse.net.ssl.OpenJSSE (referenced from: void okhttp3.internal.platform.OpenJSSEPlatform.<init>())

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:minifyReleaseWithR8'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.R8Task$R8Runnable
   > Compilation failed to complete
```

**Comment by MartinHlavna** [comment]
Thanks for PR. I will try to review and merge it as soon as possible.
